### PR TITLE
Change serverURL to be rancher.cattle-system for HA install

### PR DESCRIPTION
**Pull request by StrongMonkey**

### CODE REVIEW

**Comment by StrongMonkey** [comment]
This PR changes the serverURL in kubeconfig(Fleet uses to talk to downstream cluster) to be `rancher.cattle-system` if it detects rancher is installed in HA mode. This is an efficient way to talk to rancher API and can bypass proxy if rancher installed behind proxy.

**Comment by StrongMonkey** [comment]
Related PR https://github.com/rancher/rancher/pull/29502.

**Comment by StrongMonkey** [comment]
https://github.com/rancher/rancher/issues/29327

File: pkg/controllers/cluster/controller.go [hunk 1]
```diff
@@ -293,5 +300,26 @@ func (h *handler) getKubeConfig(cluster *v1.Cluster, status v1.ClusterStatus) (s
...
 
 func (h *handler) getServerURLAndCA() (string, string, error) {
-	return settings.GetServerURLAndCA(h.settings)
+	serverURL, ca, err := settings.GetServerURLAndCA(h.settings)
+	if err != nil {
+		return "", "", err
+	}
+
+	if dp, err := h.deploymentCache.Get(systemNamespace, "rancher"); err == nil && dp.Spec.Replicas != nil && *dp.Spec.Replicas != 0 {
+		return fmt.Sprintf("https://rancher.%s", systemNamespace), ca, nil
```
Comment (ibuildthecloud) [inline pkg/controllers/cluster/controller.go:309]: There's no way this CA is correct.

**Comment by StrongMonkey** [comment]
Changed to use internal tls port and internal ca: https://github.com/rancher/rancher/pull/29655

**Pull request merged**